lib/spec/swagger/index.js

```diff
@@ -19,16 +19,19 @@ module.exports = function (opts, cache, routes, Ref, done) {
     const swaggerObject = prepareSwaggerObject(defOpts, done)
 
     ref = Ref()
-    swaggerObject.definitions = {
+    swaggerObject.definitions = prepareSwaggerDefinitions({
       ...swaggerObject.definitions,
       ...(ref.definitions().definitions)
-    }
+    }, ref)
 
     // Swagger doesn't accept $id on /definitions schemas.
     // The $ids are needed by Ref() to check the URI so we need
     // to remove them at the end of the process
     Object.values(swaggerObject.definitions)
-      .forEach(_ => { delete _.$id })
+      .forEach(_ => {
```

Comment:
Can you move this logic inside `prepareSwaggerDefinitions`? I think it is not necessary to have two loop here.

Reply:
done - and i moved the openAPI version to the same equivalent place

Reply:
Thanks